service_configuration_lib/spark_config.py

```diff
@@ -224,12 +225,29 @@ def _append_sql_shuffle_partitions_conf(spark_opts: Dict[str, str]) -> Dict[str,
         int(spark_opts.get('spark.cores.max', 0)) or
         int(spark_opts.get('spark.executor.instances', 0)) * int(spark_opts.get('spark.executor.cores', 0))
     )
+
+    if (
+        'spark.dynamicAllocation.enabled' in spark_opts and
+        str(spark_opts['spark.dynamicAllocation.enabled']) == 'true' and
+        'spark.dynamicAllocation.maxExecutors' in spark_opts and
+        str(spark_opts['spark.dynamicAllocation.maxExecutors']) != 'infinity'
+    ):
+
+        num_partitions_dra = 2 * (
+            int(spark_opts.get('spark.dynamicAllocation.maxExecutors', 0)) *
+            int(spark_opts.get('spark.executor.cores', 0))
```

Comment:
Should this be int(spark_opts.get('spark.executor.cores', DEFAULT_EXECUTOR_CORES))?

Reply:
Makes sense. Adding it